src/hotspot/share/cds/dumpTimeClassInfo.inline.hpp

```diff
@@ -36,21 +36,15 @@
 #if INCLUDE_CDS
 
 // For safety, only iterate over a class if it loader is alive.
-// IterationHelper and DumpTimeSharedClassTable::iterate
-// must be used only inside a safepoint, where the value of
+// This function must be called only inside a safepoint, where the value of
 // k->is_loader_alive() will not change.
-template<class ITER>
-class DumpTimeSharedClassTable::IterationHelper {
-  ITER* _iter;
-public:
-  IterationHelper(ITER* iter) {
-    _iter = iter;
-  }
-  bool do_entry(InstanceKlass* k, DumpTimeClassInfo& info) {
+template<typename F>
+void DumpTimeSharedClassTable::iterate(F f) const {
+  auto g = [=] (InstanceKlass* k, DumpTimeClassInfo& info) {
```

Comment:
The style guide states we should try to only use `[&]` for capture by reference.

Reply:
Fixed.